### PR TITLE
Fix pull #17

### PR DESCRIPTION
**Pull request by ramonornela**

### CODE REVIEW

File: src/WNS.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  if (typeof args !== 'object' || !args.clientID || !args.clientSecret || !args.accessTokenPath) {
+    throw new Parse.Error(Parse.Error.PUSH_MISCONFIGURED,
+                          'WNS Configuration is invalid');
+  }
+  
+  this.clientID = args.clientID;
+  this.clientSecret = args.clientSecret;
+  this.accessTokenPath = args.accessTokenPath;
+  
+  console.log("Init WNS successful!");
```
Comment (flovilmart) [inline src/WNS.js]: Remove log

**Comment by flovilmart** [comment]
Hi Thanks for the PR!
It seems that many tests are failing as well as indentation is off at many places, can you fix that before I spend more time in code review?

Thanks!

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# http://editorconfig.org
```
Comment (flovilmart) [inline .editorconfig]: Please do not commit that file

File: .gitignore [hunk 3]
```diff
@@ -7,6 +7,7 @@ npm-debug.log*
 pids
 *.pid
 *.seed
+wpProdCert.pem
```
Comment (flovilmart) [inline .gitignore]: please remove that line

File: spec/ParsePushAdapter.spec.js [hunk 4]
```diff
@@ -33,6 +39,10 @@ describe('ParsePushAdapter', () => {
     // Check android
     var androidSender = parsePushAdapter.senderMap['android'];
     expect(androidSender instanceof GCM).toBe(true);
+	// Check windows phone
+	var winSender = parsePushAdapter.senderMap['wp'];
+    expect(winSender instanceof WNS).toBe(true);
```
Comment (flovilmart) [inline spec/ParsePushAdapter.spec.js]: Indentation is still off

File: spec/ParsePushAdapter.spec.js [hunk 5]
```diff
@@ -178,14 +214,18 @@ describe('ParsePushAdapter', () => {
...
+        deviceToken: 'wpToken'
+	  }
     ];
 
     var parsePushAdapter = new ParsePushAdapter(pushConfig);
     parsePushAdapter.send({data: {alert: 'some'}}, installations).then((results) => {
       expect(Array.isArray(results)).toBe(true);
 
-      // 2x iOS, 1x android
+      // 2x iOS, 1x android, 1x wp8.1+
```
Comment (flovilmart) [inline spec/ParsePushAdapter.spec.js]: So the result should be 4 the

File: package.json [hunk 6]
```diff
@@ -34,6 +34,7 @@
   "dependencies": {
     "apn": "^1.7.5",
     "node-gcm": "^0.14.0",
+	"wns": "^0.5.3",
```
Comment (flovilmart) [inline package.json]: indent is still off here

File: spec/ParsePushAdapter.spec.js [hunk 7]
```diff
@@ -77,12 +88,17 @@ describe('ParsePushAdapter', () => {
       {
         deviceType: 'android',
         deviceToken: undefined
-      }
+      },
+	  {
```
Comment (flovilmart) [inline spec/ParsePushAdapter.spec.js]: indentation

File: spec/ParsePushAdapter.spec.js [hunk 8]
```diff
@@ -33,6 +39,10 @@ describe('ParsePushAdapter', () => {
     // Check android
     var androidSender = parsePushAdapter.senderMap['android'];
     expect(androidSender instanceof GCM).toBe(true);
+	// Check windows phone
+	var winSender = parsePushAdapter.senderMap['wp'];
```
Comment (flovilmart) [inline spec/ParsePushAdapter.spec.js]: indentation

File: spec/ParsePushAdapter.spec.js [hunk 9]
```diff
@@ -10,6 +11,11 @@ describe('ParsePushAdapter', () => {
         senderId: 'senderId',
         apiKey: 'apiKey'
       },
+	  wp: {
```
Comment (flovilmart) [inline spec/ParsePushAdapter.spec.js]: please align with other paramterers

File: spec/ParsePushAdapter.spec.js [hunk 10]
```diff
@@ -119,7 +139,11 @@ describe('ParsePushAdapter', () => {
       {
         deviceType: 'android',
         deviceToken: undefined
-      }
+      },
+	  {
+		deviceType: 'wp',
```
Comment (flovilmart) [inline spec/ParsePushAdapter.spec.js]: please align

File: spec/ParsePushAdapter.spec.js [hunk 11]
```diff
@@ -138,6 +162,13 @@ describe('ParsePushAdapter', () => {
     expect(args[1]).toEqual([
       makeDevice('iosToken')
     ]);
+	// Check wp sender
+	expect(wpSender.send).toHaveBeenCalled();
```
Comment (flovilmart) [inline spec/ParsePushAdapter.spec.js]: indentation

File: spec/ParsePushAdapter.spec.js [hunk 12]
```diff
@@ -147,6 +178,11 @@ describe('ParsePushAdapter', () => {
         senderId: 'senderId',
         apiKey: 'apiKey'
       },
+	  wp: {
```
Comment (flovilmart) [inline spec/ParsePushAdapter.spec.js]: alignment

File: spec/ParsePushAdapter.spec.js [hunk 13]
```diff
@@ -178,14 +214,18 @@ describe('ParsePushAdapter', () => {
       {
         deviceType: 'android',
         deviceToken: undefined
-      }
+      },
+	  {
```
Comment (flovilmart) [inline spec/ParsePushAdapter.spec.js]: alignment

File: src/WNS.js [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+
+import Parse from 'parse';
+import log from 'npmlog';
+import fs from 'fs';
+import wns from 'wns';
+
+const LOG_PREFIX = 'parse-server-push-adapter WNS';
+
+function WNS(args) {
+  if (typeof args !== 'object' || !args.clientID || !args.clientSecret || !args.accessTokenPath) {
```
Comment (flovilmart) [inline src/WNS.js]: could the access token be specified as a String? or is it a certificate?

File: src/WNS.js [hunk 15]
```diff
@@ -0,0 +1,166 @@
...
+
+WNS.prototype.getNewAccessToken = function() {
+  let payload = url.format({
+    query: {
+      grant_type: 'client_credentials',
+      client_id: this.clientID,
+      client_secret: this.clientSecret,
+      scope: 'notify.windows.com'
+    }
+  }).substring(1); // strip leading ?
```
Comment (flovilmart) [inline src/WNS.js]: why?

File: src/WNS.js [hunk 16]
```diff
@@ -0,0 +1,166 @@
...
+          });
+        });
+
+        return Parse.Promise.when(promises);
+      }
+    }
+  });
+}
+
+WNS.prototype.getNewAccessToken = function() {
```
Comment (flovilmart) [inline src/WNS.js]: this method will always return undefined

File: src/WNS.js [hunk 17]
```diff
@@ -0,0 +1,166 @@
...
+          log.error(LOG_PREFIX, `cannot get currentAccessToken From WNS`);
+        } else {
+          fs.writeFile(this.accessTokenPath, currentAccessToken, function(err) {
+            if (err) {
+              log.error(LOG_PREFIX, `cannot write new currentAccessToken to %s`,this.accessTokenPath);
+            }
+          });	
+        }
+
+        let wnsPayload = getWNSToastPayload(data.title, data.alert);
```
Comment (flovilmart) [inline src/WNS.js]: alignment

File: src/ParsePushAdapter.js [hunk 18]
```diff
@@ -12,7 +13,7 @@ export class ParsePushAdapter {
   supportsPushTracking = true;
 
   constructor(pushConfig = {}) {
-    this.validPushTypes = ['ios', 'android'];
+    this.validPushTypes = ['ios', 'android','wp'];
```
Comment (TylerBrock) [inline src/ParsePushAdapter.js]: space after comma (between arguments)

**Comment by flovilmart** [comment]
@ramonornela  Can you try to fix all the issues in one commit please? and rebase as your branch is out of date.

**Comment by ramonornela** [comment]
@flovilmart ok one commit fix identation and other of refactoring method 'sendWNSNow'.

**Comment by ramonornela** [comment]
@flovilmart see #26

**Pull request closed without merging**